### PR TITLE
Add date selection on needs index

### PR DESCRIPTION
**Pull request by joelmichael**
Adds a new react component.
Renames the class in NewNeedDatePicker so they are distinguishable.
Adds a scope to Need.
Updates index method to handle presence of a start date.
Styling not completed yet.
Currently has an issue with the date being offset by one due to time zone.

### CODE REVIEW

**Comment by joelmichael** [comment]
Implements #192

File: app/views/needs/index.html.haml [hunk 1]
```diff
@@ -3,6 +3,9 @@
 
 .grid-x
   .cell.medium-7.large-9.small-order-1.medium-order-2
+    - if current_user.role == 'admin'
```
Comment (benjaminwood) [inline app/views/needs/index.html.haml]:
```suggestion
    - if current_user.admin?
```

File: app/controllers/needs_controller.rb [hunk 2]
```diff
@@ -4,7 +4,15 @@ class NeedsController < ApplicationController
...
-    @needs = policy_scope(Need).includes(:shifts).current.order(:start_at)
+    @needs = policy_scope(Need).includes(:shifts).order(:start_at)
+
+    if current_user.role == 'admin' && params[:date]
+      @date = Date.parse(params[:date])
+      @needs = @needs.on_date(@date)
+    else
+      @date = nil
+      @needs = @needs.current
+    end
```
Comment (benjaminwood) [inline app/controllers/needs_controller.rb]: If the `date` param was an invalid string (for example, `abc`), `Date.parse` will blow up with an `ArgumentError`. What if we did something like this?

```ruby
def index
  authorize Need

  needs_scope = policy_scope(Need).includes(:shifts).order(:start_at)

  @date = safe_parse_date(params[:date])
  @needs = @date && current_user.admin? ? needs_scope.on_date(@date) : needs_scope.current
end

private

def safe_parse_date(date)
  Date.parse(date)
rescue ArgumentError, TypeError
end
```

Comment (benjaminwood) [reply app/controllers/needs_controller.rb]: Could also write `safe_parse_date` like this:


```ruby
def safe_parse_date(date)
  date && Date.parse(date)
rescue ArgumentError
end
```

It makes it a bit more clear what is going on.

Comment (joelmichael) [reply app/controllers/needs_controller.rb]: While true, there should be no legitimate reason that a user would be manually inputting anything other than the date format from the calendar there. I think raising an exception is probably an acceptable outcome of that situation, but can change this if you prefer.

Comment (benjaminwood) [reply app/controllers/needs_controller.rb]: Happy to leave it to your discretion. Please do use current_user.admin? here as well, though.

**Pull request merged**